api/nvidia.com/resource/v1beta1/computedomain.go

```diff
@@ -67,8 +67,9 @@ type ComputeDomainChannelSpec struct {
 	// Allows for requesting all IMEX channels (the maximum per IMEX domain) or
 	// precisely one.
 	// +kubebuilder:validation:Enum=All;Single
-	// +kubebuilder:default=Single
-	AllocationMode string `json:"allocationMode"`
+	// +kubebuilder:default:=Single
+	// +kubebuilder:validation:Optional
```

Comment:
Reading up on things in https://book.kubebuilder.io/reference/markers/crd-validation and other resources, I think I now understand that 
- `default:=` is the correct syntax (not just `default=`)
- `default:=` always needs to go hand in hand with `Optional`, otherwise it practically a noop

Reply:
Technically `+kubebuilder:validation:Optional` is not necessary if you have `omitempty`, but it's fine to keep ir for better readibility -- and in this case (unlike with `Index`, we do want to keep `omitempty` here).